compendium/DeclarativeServices/src/metadata/MetadataParserImpl.cpp

```diff
@@ -135,6 +138,81 @@ MetadataParserImplV1::CreateComponentMetadata(const AnyMap& metadata) const
   ObjectValidator(metadata, "name", /*isOptional=*/true)
     .AssignValueTo(compMetadata->name);
 
+  // component.configuration-policy (Optional)
+  compMetadata->configurationPolicy = CONFIG_POLICY_IGNORE;
+  unsigned short configPolicy = 0;
+  object =
+    ObjectValidator(metadata, "configuration-policy", /*isOptional=*/true);
+  if (object.KeyExists()) {
+    object.AssignValueTo(compMetadata->configurationPolicy);
+    configPolicy = 1;
+  }
+
+  // component.configuration-pid (Optional)
+  unsigned short configPid = 0;
+  object = ObjectValidator(metadata, "configuration-pid", /*isOptional=*/true);
+  if (object.KeyExists()) {
+    configPid = 1;
+    if (configPolicy == 1) {
```

Comment:
Kevin: Mike would've probably brought up that having the variable on the right-side of == minimizes chance for assignment error